test-harness/specs/E2E-Accept-Content-Negotiation-AC14.oas2.txt

```diff
@@ -0,0 +1,39 @@
+====test====
+Sending a GET request with no Accept header specified returns
+the first example - content in application/xml.
+====spec====
+swagger: '2.0'
+  version: 1.0.0
+paths:
+  "/pet/{petId}":
+    get:
+      produces:
+      - application/xml
+      - application/json
+      parameters:
+      - name: petId
+        in: path
+        description: ID of pet to return
+        required: true
+        type: integer
+        format: int64
+      responses:
+        '200':
+          schema:
+            type: object
+            properties:
+              id:
+                type: integer
+                format: int64
+              name:
+                type: string
+                example: doggie
+====server====
+mock -p 4010
+====command====
+curl -i http://localhost:4010/pet/123
+====expect====
+HTTP/1.1 200 OK
+content-type: application/xml
+
+<id>0</id><name>doggie</name>
```

Comment:
This is not valid XML, how are we generating it? We always need a root element of some sort. 

There is nothing in the spec about how to create a XML representation from basic data like this, but [a reasonable example](https://swagger.io/docs/specification/data-models/representing-xml/) from the folks over at SmartBear is to wrap it with a XML tag which matches the name of the model (if its a model).

If not... probably something arbitrary. I'd rather we shoved a `<xml>` tag and let folks configure it later when we add full XML keyword support (#456), as that beats outputting invalid XML.

Reply:
We're using an XML serialized; I'm quite sure I can configure it to have a root tag.

Reply:
We're using an XML Serializer library found on NPM; I'm going to try to check if it can auto-enclose the payload for us.